### PR TITLE
ORC: Use MetricsConfig

### LINKED ISSUE #1268: Respect MetricsConfig while importing ORC data
We should respect `MetricsConfig` in `SparkTableUtil`.

```
// TODO: use MetricsConfig in listOrcPartition
return listOrcPartition(partition, uri, spec, conf);
```

### PR DESCRIPTION
**Pull request by edgarRd**
Unlike Parquet metrics, ORC was not using `MetricsConfig` to let the user configure which metrics to collect or truncate string column metrics.

This PR fixes #1268 by adding `MetricsConfig` support in ORC metrics. I've also added unit tests to `TestMetrics`, which runs for both Parquet and ORC, to check the different `MetricsModes` and the truncation of string and binary (not supported in ORC) columns.

PTAL @aokolnychyi @rdsr @shardulm94 Thanks!

### CODE REVIEW

**Comment by edgarRd** [comment]
Friendly ping in this PR: @aokolnychyi @rdsr @rdblue Thanks!

File: orc/src/main/java/org/apache/iceberg/orc/OrcMetrics.java [hunk 1]
```diff
@@ -203,7 +227,29 @@ private static Metrics buildOrcMetrics(final long numOfRows, final TypeDescripti
       BooleanColumnStatistics booleanStats = (BooleanColumnStatistics) columnStats;
       max = booleanStats.getTrueCount() > 0;
     }
-    return Optional.ofNullable(Conversions.toByteBuffer(column.type(), max));
+    return Optional.ofNullable(Conversions.toByteBuffer(type, truncateIfNeeded(Bound.UPPER, type, max, metricsMode)));
+  }
+
+  private static Object truncateIfNeeded(Bound bound, Type type, Object value, MetricsMode metricsMode) {
+    // ORC only stores min/max for String columns
+    if (value == null || metricsMode == MetricsModes.Full.get() || type.typeId() != Type.TypeID.STRING) {
```
Comment (shardulm94) [inline orc/src/main/java/org/apache/iceberg/orc/OrcMetrics.java:235]: Might be better to check if the Metrics mode != Truncate, seems easier to read that way. Also we have a Precondition later to assert the same, which may be unnecessary if we make the check here.

Comment (edgarRd) [reply orc/src/main/java/org/apache/iceberg/orc/OrcMetrics.java]: Done.

**Comment by edgarRd** [comment]
Thanks for the review @shardulm94 - I've updated the PR.

**Comment by edgarRd** [comment]
@rdsr @aokolnychyi PTAL - do you have any comments?

**Comment by shardulm94** [comment]
Since this has been open for additional comments for a while now I am merging this. We can handle any followups in a separate PR. Thanks @edgarRd for the PR!

**Comment by edgarRd** [comment]
Thanks @shardulm94 !

**Pull request merged**